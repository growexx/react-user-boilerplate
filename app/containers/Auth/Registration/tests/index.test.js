/**
 *
 * Tests for Registration
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import history from 'utils/history';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import configureStore from '../../../../configureStore';
import { Registration } from '../index';

let store;
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <Registration />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<Registration />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
});
