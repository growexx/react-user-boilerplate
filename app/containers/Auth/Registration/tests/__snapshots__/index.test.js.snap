// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> Should render and match the snapshot 1`] = `
.c1 {
  font-family: 'Montserrat',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
}

.c1 .createAccount {
  margin: 0;
  text-align: center;
  font-size: 42px;
  font-weight: 700;
  color: #4d186e;
}

.c1 .emailRegistration {
  margin-top: 31px;
  color: rgba(0,0,0,0.85);
  font-weight: 400;
  font-size: 14px;
  text-align: center;
}

.c1 .registrationSubContainer {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .socialIcons {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .socialIcons > span {
  border: 1px solid #d9d9d9;
  padding: 12px;
  border-radius: 50%;
  margin: 12px;
}

.c1 .accountData input {
  margin-top: 12px;
  background-color: #f5f5f5;
}

.c1 .accountData input[value] {
  color: rgba(0,0,0,0.45);
  font-size: 15px;
}

.c1 .accountData.input-margin-0 input {
  margin-top: 0;
}

.c1 .accountData .ant-input-password {
  background-color: #f5f5f5;
}

.c1 button {
  margin-top: 31px;
  background: #4d186e;
  border: 2px solid #fff;
  border-radius: 50px;
  color: #fff;
  text-align: center;
  font-size: 15px;
  font-weight: 500;
  height: 40px;
  width: 160px;
}

.c1 button:hover {
  background: #4d186e;
  border: 2px solid #fff;
  color: #fff;
}

.c1 button:focus {
  background: #4d186e;
  border: 2px solid #fff;
  color: #fff;
}

.c0 {
  font-family: 'Montserrat',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 .sideContainer {
  max-width: 513px;
  width: 100%;
  background: #4d186e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 32px;
}

.c0 .sideContainer .registrationSideContainer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
}

.c0 .sideContainer .registrationSideContainer .title {
  text-align: center;
  font-size: 42px;
  font-weight: 700;
  margin: 0;
}

.c0 .sideContainer .registrationSideContainer .subTitle {
  width: 60%;
  text-align: center;
  margin: 0;
}

.c0 .sideContainer .registrationSideContainer button {
  margin-top: 35px;
  height: 40px;
  width: 160px;
  background: transparent;
  border: 2px solid #fff;
  border-radius: 50px;
  text-align: center;
  color: #fff;
}

.c0 .sideContainer .registrationSideContainer button:hover {
  color: #fff;
}

@media only screen and (max-width:767px) {
  .c1 .registrationSubContainer {
    width: 80%;
  }
}

<div
  class="c0"
>
  <div
    class="sideContainer"
  >
    <div
      class="registrationSideContainer"
    >
      <p
        class="title"
      >
        <span>
          Welcome Back!
        </span>
      </p>
      <p
        class="subTitle"
      >
        <span>
          To keep connected with us, please log in with your personal info
        </span>
      </p>
      <a
        href="/login"
      >
        <button
          class="ant-btn"
          type="button"
        >
          <span>
            SIGN IN
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    class="c1"
  >
    <p
      class="createAccount"
    >
      <span>
        Create Account
      </span>
    </p>
    <div
      class="registrationSubContainer"
    >
      <p
        class="emailRegistration"
      >
        <span>
          or use your email for registration
        </span>
      </p>
      <div
        class="accountData input-margin-0"
      >
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="username"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="email"
                    placeholder="email"
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-affix-wrapper ant-input-password"
                  >
                    <input
                      action="click"
                      class="ant-input"
                      id="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        class="anticon anticon-eye-invisible ant-input-password-icon"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <button
        class="ant-btn"
        type="button"
      >
        <span>
          SIGN UP
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<Registration /> Should render and match the snapshot Loadable 1`] = `null`;
